lxd/storage/drivers/generic_vfs.go

```diff
@@ -682,7 +684,19 @@ func genericVFSBackupUnpack(d Driver, vol Volume, snapshots []string, srcData io
 			// Extract filesystem volume.
 			d.Logger().Debug(fmt.Sprintf("Unpacking %s filesystem volume", volTypeName), log.Ctx{"source": srcPrefix, "target": mountPath, "args": args})
 			srcData.Seek(0, 0)
-			err = shared.RunCommandWithFds(r, nil, "tar", args...)
+
+			f, err := os.OpenFile(mountPath, os.O_RDONLY, 0)
+			if err != nil {
+				return errors.Wrapf(err, "Error opening directory")
+			}
+			defer f.Close()
+
+			algName := ""
```

Comment:
What is algName. I'm struggling to understand what the `unpacker` arg of `ExtractWithFds` is? Is that not the same as `cmd`? It doesn't seem like we have the right naming and/or abstraction here yet.

Reply:
It's a name of compression algorithm e.g gzip or bzip2. Apparmor needs this two in profile when `tar` with compression is called. So for example we need to have:
```
/usr/bin/tar mixr,
/usr/bin/gzip mixr,
```
As you said maybe naming is not right in this case or maybe we should pass list of binaries to apparmor which should be allowed to run.

Reply:
Ah I see so its allowed commands that the `cmd` itself can run (in addition to itself).

Why don't we have this argument be a []string and be called `allowedCmds` or something like that to show its relationship to the existing `cmd` argument (and explain it in the function description).

The the `cmd` argument itself can be added to that list internally and passed to the apparmor profile as a single list of allowed commands?

Reply:
I've added `allowedCmds []string` to `ExtractWithFds` function.